packages/ui/src/SelectField.tsx

```diff
@@ -0,0 +1,145 @@
+    }
+  | {
+      isClearable?: false
+      value: SelectFieldOption<V>
+      onChange: (newValue: SelectFieldOption<V>) => void
+    }
+export type SelectFieldExtraProps<V> = {
+  options: SelectFieldOption<V>[]
+  label: string
+  required?: boolean
+  helperText?: string | ReactElement
+  className?: string
+  errorClassName?: string
+  placeholder?: string
+  menuPortalTarget?: 'body' | 'self' | HTMLElement | null
+} & DataTestProp &
+  Pick<InputHTMLAttributes<HTMLElement>, 'autoFocus' | 'disabled' | 'autoComplete'> &
+  Pick<ReactSelectProps, 'isSearchable' | 'menuIsOpen' | 'menuPlacement'>
+
+export type SelectProps<V> = SelectFieldCoreStaticProps & SelectFieldCoreDynamicProps<V> & SelectFieldExtraProps<V>
+
+const getMenuContainer = (menuPortalTarget: 'body' | 'self' | HTMLElement | null): HTMLElement | null | undefined => {
+  if (menuPortalTarget == 'body') {
+    return document.body
+  }
+  if (menuPortalTarget == 'self') {
+    return undefined
+  }
+  return menuPortalTarget
+}
+
+export const SelectField = <V,>({
+  'data-test': dataTest,
+  className,
+  error,
+  errorClassName,
+  isClearable,
+  disabled,
+  isSearchable = false,
+  label,
+  name,
+  required,
+  value,
+  onChange,
+  helperText,
+  menuPortalTarget = 'body',
+  ...rest
+}: SelectProps<V>): ReactElement<SelectProps<V>> => {
+  const idRef = useRef(uuid())
+
+  useIsomorphicLayoutEffect(() => {
+    const menuContainer = getMenuContainer(menuPortalTarget)
+
+    if (menuContainer && !menuContainer.className.includes(styles.menuContainer)) {
+      menuContainer.className = `${menuContainer.className} ${styles.menuContainer}`
+    }
+  }, [menuPortalTarget])
+
+  return (
+    <div
+      data-test={dataTest}
+      className={cn(
+        styles.container,
+        {
+          [styles.disabled]: disabled,
+          [styles.hasError]: error,
+          [styles.empty]: !value,
+        },
+        // Menu container is either this select itself or document.body
+        // We can always add this class to the select itself because even if the menu container is body it won't break it
+        // However, if it is the select itself, it will properly add necessary specificity level to the menu styles.
+        styles.menuContainer,
+        className,
+      )}
+    >
+      <Label id={idRef.current} label={label} />
+      <div className={styles.selectBlock}>
+        <ReactSelect<SelectFieldOption<V>>
+          inputId={idRef.current}
+          className="hz-select-field"
+          classNamePrefix="hz-select-field"
+          // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-invalid_attribute
+          aria-errormessage={error && errorId(idRef.current)}
+          aria-invalid={!!error}
+          aria-required={required}
+          isClearable={isClearable ?? false}
+          isDisabled={disabled}
+          isMulti={false}
+          isSearchable={isSearchable}
```

Comment:
I would love to see multi select and searchable select stories.

Reply:
Totally forgot about searchable select. Thnx. As to multiselect. I think it should be a different component.

Reply:
Addressed in https://github.com/hazelcast/frontend-shared/pull/72/commits/acbad02b1ebd440112f7c2ef53b08a90a6b49557

Reply:
@aigoncharov So we're going to use `SelectField` as a base for `MultiSelectField` component, right?

Reply:
@danad02 currently, `SelectField` does not accept `isMulti` as a prop, so you won't be able to do it unless you change the code :) I just feel like MultiSelect has a completely different look and feel, so I would rather do it as a standalone component.

Reply:
Ah got it :smile: Well there is no information about it in the style guide. But if the future designs tell us to use this as a multiselect then there is nothing easier than enabling it and change styles a bit. Approved!

Reply:
@danad02 I think we will implement MultiSelect based on the same react-select. I just do not want to do with lots of conditional code to make it happen. I would rather copy-paste something from this component into a new shiny `MultiSelect` and make it works there for the multi select use-case only

Reply:
Yeah there would probably be a surprising number of conditions :+1: